### PR TITLE
workflows: use `--skip-online-checks`

### PR DESCRIPTION
**Pull request by carlocab**
This follows from the discussion at Homebrew/homebrew-test-bot#699.

### CODE REVIEW

**Review by SMillerDev — DISMISSED** [review]
😍

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -138,6 +138,13 @@ jobs:
               console.log('No CI-build-dependents-from-source label found. Not passing --build-dependents-from-source to brew test-bot.')
             }
 
+            if (label_names.includes('CI-skip-online-checks')) {
```
Comment (MikeMcQuaid) [inline .github/workflows/tests.yml:141]: This makes me uncomfortable and is scope creep on what's discussed before. When would we pass this?

**Review by MikeMcQuaid — COMMENTED** [review]
👍🏻  on build bottle/rebottle. 👎🏻 on tests.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (carlocab) [reply .github/workflows/tests.yml:141]: This is case 3 from https://github.com/Homebrew/homebrew-test-bot/pull/699. In particular, I don't think we can merge #89042 without it.

Comment (MikeMcQuaid) [reply .github/workflows/tests.yml:141]: I am concerned this will be misused to ignore rather than fix these sort of checks. If we have annotations that produce warnings on PR: 👍🏻, otherwise I'm still 👎🏻 on this.

Comment (MikeMcQuaid) [reply .github/workflows/tests.yml:141]: (this feels like a slippery slope that ends up with `CI-skip-failing-tests`)

Comment (carlocab) [reply .github/workflows/tests.yml:141]: I'm sympathetic to your concerns, but:

1. if we're contemplating just getting rid of these tests, it surprises me that we would be this worried about ignoring and not fixing them.
2. if a maintainer is set on ignoring failing CI, then there's not much we can do about that. This change just makes it so that, when that happens, we're at much less likely to be flying blind with respect to how much is broken by that decision.

That said, I'm fine with waiting until we can implement annotations that produce warnings.

**Comment by carlocab** [comment]
I'll split out the bottle/rebottle changes to a separate PR.

**Comment by carlocab** [comment]
#89268

File: .github/workflows/tests.yml (hunk 1 above)
Comment (MikeMcQuaid) [reply .github/workflows/tests.yml:141]: > 1. if we're contemplating just getting rid of these tests, it surprises me that we would be this worried about ignoring and not fixing them.

I think deleting is much preferable to ignoring. 

> 2\. if a maintainer is set on ignoring failing CI, then there's not much we can do about that. This change just makes it so that, when that happens, we're at much less likely to be flying blind with respect to how much is broken by that decision.

I think merging red CI or merging green CI with warnings is different to merging green CI which only passed because of a particular label.

> That said, I'm fine with waiting until we can implement annotations that produce warnings.

❤️ thanks!

Comment (MikeMcQuaid) [reply .github/workflows/tests.yml:141]: > That said, I'm fine with waiting until we can implement annotations that produce warnings.

Also, FYI, it seems to be just the matter of adding a `puts` with the relevant output:
https://github.com/Drieam/rspec-github/blob/c00ee4b3ba3051fc1e07b084c0f4ea544b455736/lib/rspec/github/formatter.rb#L18-L22

Comment (carlocab) [reply .github/workflows/tests.yml:141]: Thanks for the hint -- I was wondering if that was all there is to it.

**Comment by SMillerDev** [comment]
I think the work your doing here is a good first step towards splitting out all the online checks to a separate runner. That way we can do the build checks on the OS we want to test, and move the general metadata tests to a separate step that will finish quicker and not-block builds.

**Comment by MikeMcQuaid** [comment]
> I think the work your doing here is a good first step towards splitting out all the online checks to a separate runner.

Yes, I like the idea of doing this 👍🏻.

**Comment by carlocab** [comment]
Here's a different idea: why don't we skip online checks for `test-bot` runs where `@testing_formulae.count > 5`? It's a bit of an arbitrary line, but, in my experience, these tests are really more trouble than they're worth when a PR has to modify at least that many formulae (and most changes are almost always just `revision` bumps that don't touch anything online-related).

**Comment by SMillerDev** [comment]
That sounds reasonable to me

**Comment by carlocab** [comment]
Homebrew/homebrew-test-bot#704

**Comment by MikeMcQuaid** [comment]
Yeh, great idea 👍🏻
